lib/pbench/cli/server/__init__.py

```diff
@@ -30,8 +31,8 @@ def config_setup(context: object, name: str) -> None:
 
     # We're going to need the Postgres DB to track dataset state, so setup
     # DB access.
-    try:
-        if not Database.db_session:
+    if Database.db_session is None:
+        try:
             Database.init_db(config, logger)
-    except Exception:
-        Database.init_db(config, logger)
+        except Exception:
+            logger.exception("Exception while initializing sqlalchemy database")
```

Comment:
What do you mean?

Reply:
Why would we find `Database.db_session` non-`None` here? Can this happen in server startup, too? I suppose this _is_ a common point for Click-based CLI commands, at least, but is that enough?

Reply:
yeah I see what you mean, I had to do this because of the following reasons
1. Each click user management command `user-create`/`user-delete`/... uses this function to setup a db access.
2. In the multiple unit test we are using more than one cli command for example to test `user-delete` I have to use `user-create` first. Therefore without checking `Database.db_session`  for None these 2 commands will spin up 2 different instances of in-memory databases and we cant have that, it is pointless, we need one db instance where 2 commands interact on the data.

Reply:
This feels like you're working around a problem which is perhaps the result of reusing code improperly.

Having the `user-delete` test reuse the code from the `user-create` test is great, but, obviously, `user-delete` shouldn't be using _all_ the code, because you run into the database re-creation problem.  Instead, I think you need to separate the database creation into a PyTest fixture, such that the database gets created only once.  Also, it's likely that `user-delete` only wants to invoke part of `user-create` anyway, in which case that part should be refactored into a subroutine which they can safely share without `user-delete` inheriting other constraints and invariants from the `user-create` implementation.